workers/docker-worker/src/lib/features/chain_of_trust.js

```diff
@@ -41,7 +41,7 @@ class ChainOfTrust {
   }
 
   async killed(task) {
-    if (task.isCanceled()) {return;}
+    if (task.isCanceled() || task.isAborted()) {return;}
```

Comment:
Currently if a task is aborted (e.g. because it is not completed within `maxRunTime`), the certified log and other chain of trust artifacts will not be uploaded, so it will make verifying their integrity more difficult. Is this definitely desirable?

@escapewindow Do you care whether timed-out tasks lose their certified logs and chain of trust certificate etc, or do you only care about chain of trust artifacts for successful tasks?

Reply:
I see the issue in #3525 now, and realise this fix was motivated by failing worker-shutdowns rather than task timeouts (which I believe are also considered task abortions).

It looks like the error message in #3525 is due to [this line](https://github.com/taskcluster/taskcluster/blob/5a25a717299e9eae61d7ed0935ceb028b8319a26/workers/docker-worker/src/lib/features/artifacts.js#L207) not having run at the time the task is aborted, so I wonder if that initialisation just needs to happen sooner?

Reply:
> With this change, if a task is aborted (e.g. because it is not completed within `maxRunTime`), the certified log and other chain of trust artifacts will not be uploaded, which would make verifying their integrity more difficult. Is this definitely desirable?
> 
> @escapewindow Do you care whether timed-out tasks lose their certified logs and chain of trust certificate etc, or do you only care about chain of trust artifacts for successful tasks?

Currently, and for the foreseeable future, we only care about CoT for successful tasks. I don't know if we can trust artifacts from unsuccessful tasks for release purposes in general.

Reply:
My logic was that if we're not doing CoT for cancelled tasks, then aborted tasks do not need it either.  And the current behavior is for an aborted task to fail with the error described in #3525 and without CoT artifacts, so this isn't changing current behavior -- just avoiding logging an unnecessary error.